verl/utils/fs.py

```diff
@@ -290,3 +299,78 @@ def local_mkdir_safe(path):
         os.makedirs(path, exist_ok=True)
 
     return path
+
+
+def copy_local_path_from_remote(
+    src: str,
+    cache_dir=None,
+    filelock=".file.lock",
+    verbose=False,
+    recursive: Optional[bool] = None,
+    local_path: Optional[str] = None,
+) -> str:
+    """
+    Used to download files from a remote source (S3 or HDFS).
+
+    Deprecated. Please use copy_to_local which calls this function.
+    """
+    from filelock import FileLock
+
+    from verl.utils.s3_io import bulk_download, file_download, parse_uri
+
+    assert src[-1] != "/", f"Make sure the last char in src is not / because it will cause error. Got {src}"
+
+    if is_non_local(src):
+        # download from hdfs to local
+        if cache_dir is None:
+            # get a temp folder
+            cache_dir = tempfile.gettempdir()
+        os.makedirs(cache_dir, exist_ok=True)
+        assert os.path.exists(cache_dir)
+        if not local_path:
+            local_path = get_local_temp_path(src, cache_dir)
+        # get a specific lock
+        filelock = md5_encode(src) + ".lock"
+        lock_file = os.path.join(cache_dir, filelock)
+        with FileLock(lock_file=lock_file):
+            if not os.path.exists(local_path):
+                if verbose:
+                    print(f"Copy from {src} to {local_path}")
+                if src.startswith(_S3_PREFIX):
+                    from verl.utils.s3_io import bulk_download, file_download, parse_uri
+
+                    bucket, key_or_prefix, recursive = parse_uri(src, is_dir=recursive)
+
+                    if recursive:
+                        bulk_download(bucket, key_or_prefix, local_path)
+                    else:
+                        file_download(bucket, key_or_prefix, local_path)
+                else:
+                    copy(src, local_path)
+        return local_path
+    else:
+        return src
+
+
+def upload_local_file_to_s3(
```

Comment:
what about making the function name more general, like `copy_to_remote` and support s3 path for now